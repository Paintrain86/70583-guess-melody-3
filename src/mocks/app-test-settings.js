const settings = {
  luckText: `Тестовый текст на удачу`,
  gameTime: 50,
  mistakesCount: 2
};

const questions = [
  {
    type: `genre`,
    genre: {
      name: `pop`,
      displayName: `Поп`
    },
    answers: [
      {
        id: 1,
        src: `song.mp3`,
        genre: `pop`
      },
      {
        id: 2,
        src: `song.mp3`,
        genre: `rap`
      }
    ]
  },
  {
    type: `artist`,
    song: {
      src: `song.mp3`,
      artist: `Test`
    },
    answers: [
      {
        picture: `pic.jpeg`,
        artist: `Test`
      },
      {
        picture: `pic.jpeg`,
        artist: `Test 2`
      },
      {
        picture: `pic.jpeg`,
        artist: `Test 3`
      }
    ]
  }
];

const appParams = {
  gameSettings: settings,
  gameQuestions: questions
};

export {settings, questions, appParams};
