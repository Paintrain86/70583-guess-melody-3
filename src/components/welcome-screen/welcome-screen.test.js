import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from './welcome-screen.jsx';

import {settings} from '../../mocks/app-test-settings.js';

describe(`WelcomeScreen`, () => {
  it(`Should be rendered correctly`, () => {
    const onStartClick = jest.fn();

    const markup = renderer
      .create(<WelcomeScreen
        {...settings}
        onStartClick={onStartClick}
      />)
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
