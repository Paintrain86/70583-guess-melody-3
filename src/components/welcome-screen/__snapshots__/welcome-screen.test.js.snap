// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen Should be rendered correctly 1`] = `
Array [
  <ul>
    <li>
      <strong>
        Текст удачи:
      </strong>
       
      Тестовый текст на удачу
    </li>
    <li>
      <strong>
        Время на прохождение:
      </strong>
       
        50 секунд
    </li>
    <li>
      <strong>
        Максимальное кол-во ошибок:
      </strong>
       
      2
    </li>
  </ul>,
  <button
    onClick={[Function]}
  >
    Погнали
  </button>,
]
`;
